src/engine/image.cpp

```diff
@@ -993,7 +993,7 @@ namespace fheroes2
         const int32_t widthOut = out.width();
 
         if ( flip ) {
-            const int32_t offsetInY = inY * widthIn + widthIn - 1 - inX;
+            const int32_t offsetInY = inY * widthIn + width - 1 + inX;
```

Comment:
The same as for 793 line:
```suggestion
            const int32_t offsetInY = inY * widthIn + widthIn - 1 - inX;
```